src/main/webapp/consentGroupDocuments/ConsentGroupDocuments.js

```diff
@@ -65,6 +67,20 @@ class ConsentGroupDocuments extends Component {
     });
   };
 
+  getUseRestriction = () => {
+    ConsentGroup.getUseRestriction(this.props.useRestrictionUrl, this.props.projectKey).then(resp => {
+      this.setState(prev => {
+        prev.restriction = resp.data.restriction;
+        if (resp.data.restrictionId !== null) {
+          prev.restrictionId = resp.data.restrictionId.id;
+        } else {
+          prev.restrictionId = null;
+        }
+        return prev;
+      })
```

Comment:
I think it would be a lot safer to take this if out of the setState call. You have `resp.data.restrictionId` in `resp`, so something like this might work:
```
const newRestrictionId = resp.data.restrictionId ? resp.data.restrictionId.id : null;
this.setState(prev => { 
    prev.restrictionId = newRestrictionId; 
    return prev;
});
```